src/mpid/ch4/shm/xpmem/xpmem_pre.h

```diff
@@ -14,6 +14,41 @@
 #include <mpi.h>
 #include <xpmem.h>
 
+#define RLOW_LOCATE_MIN_RIGHT(node, rlow) (rlow >= node->min)
+#define RHIGH_LOCATE_MAX_LEFT(node, rhigh) (rhigh <= node->max)
+#define RLOW_LOCATE_LOW_LEFT(node, rlow) (rlow < node->low)
+#define RHIGH_LOCATE_HIGH_LEFT(node, rhigh) (rhigh <= node->high)
+
```

Comment:
If these macros are used only in the xpmem_avl file, please move them there. Besides, you might need add namespace prefix since they are defined in a header file.